intro_to_ada.rst

```diff
@@ -3479,32 +3479,802 @@ Predefined exceptions
+       end T2;
+
+    begin
+       Put_Line ("In main");
+    end Show_Simple_Tasks;
 
 Simple synchronization
-----------------------
+~~~~~~~~~~~~~~~~~~~~~~
+
+As we've just seen, as soon as the main application starts, its tasks
+also start automatically. The main application will continue its
+processing until it reaches the end of its implementation. At this point,
+however, it will not finish. Instead, the main application will wait until
+its tasks have finished before it finishes itself. In other words, after
+this waiting process, a synchronization between the main application and
+its tasks occurs. After this final synchronization, the main application
+may finish. For example:
+
+.. code-block:: ada
+
+    --% run_file: Show_Simple_Sync.adb
+    --% make_flags: -gnaty
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    procedure Show_Simple_Sync is
+       task T;
+       task body T is
+       begin
+          for I in 1 .. 10 loop
+             Put_Line ("hello");
+          end loop;
+       end;
+    begin
+       null;
+       --  Will wait here until all tasks have terminated
+    end Show_Simple_Sync;
+
+A synchronization is also achieved if we move the task to a separate
+package. In the example below, we declare a task ``T`` in the package
+``Simple_Sync_Pkg``.
+
+.. code-block:: ada
+
+    --% src_file: Simple_Sync_Pkg.ads
+    --% cflags: -gnaty
+
+    package Simple_Sync_Pkg is
+       task T;
+    end Simple_Sync_Pkg;
+
+This is the corresponding package implementation:
+
+.. code-block:: ada
+
+    --% src_file: Simple_Sync_Pkg.adb
+    --% cflags: -gnaty
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    package body Simple_Sync_Pkg is
+       task body T is
+       begin
+          for I in 1 .. 10 loop
+             Put_Line ("hello");
+          end loop;
+       end T;
+    end Simple_Sync_Pkg;
+
+As soon as the package is :ada:`with`'ed for the main procedure, the task
+``T`` defined in the package will be part of the main application. For
+example:
+
+.. code-block:: ada
+
+    --% run_file: Test_Simple_Sync_Pkg.adb
+    --% make_flags: -gnaty
+
+    with Simple_Sync_Pkg;
+
+    procedure Test_Simple_Sync_Pkg is
+    begin
+       null;
+       --  Will wait here until all tasks have terminated
+    end Test_Simple_Sync_Pkg;
+
+Again, as soon as the main application reaches its end, it will
+synchronize with task ``T`` from ``Simple_Sync_Pkg`` before finishing.
 
 Delay
------
+~~~~~
+
+A delay may be introduced by using the keyword :ada:`delay`. This will
+put the task to sleep for the amount of seconds specified in the delay
+statement. For example:
+
+.. code-block:: ada
+
+    --% run_file: Show_Delay.adb
+    --% make_flags: -gnaty
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    procedure Show_Delay is
+
+       task T;
+
+       task body T is
+       begin
+          for I in 1 .. 10 loop
+             Put_Line ("hello");
+             delay 1.0;
+             --    ^ Wait 1.0 seconds
```

Comment:
It would be interesting here to add a delay in the main, to showcase the fact that the main task is just an implicit task, I think :)